iothub/device/devdoc/architecture.md

```diff
@@ -2,43 +2,32 @@
 
 ## Abstract
 
-This document describes the internal architecture for the Azure IoT Hub Device SDK. Applications use `DeviceClient` and `ModuleClient` which represent the two types of device identities. The objects are internally represented by the same `InternalClient` class that is built by `ClientFactory`. During object creation, a transport pipeline, following the _chain of responsibility_ pattern, is also created. All but the last element of the pipeline will perform alterations of application requests. The last element, based on `TransportHandler`, has the responsibility of converting IoT Hub request messages into the selected protocol and perform network I/O.
+This document describes the internal architecture for the Azure IoT Hub Device SDK. Applications use `IotHubDeviceClient` and `IotHubModuleClient` which represent the two types of device identities. During object creation, a transport pipeline, following the _chain of responsibility_ pattern, is also created. All but the last element of the pipeline will perform alterations of application requests. The last element, based on `TransportHandlerBase`, has the responsibility of converting IoT Hub request messages into the selected protocol and perform network I/O.
```

Comment:
I'd argue we don't use the chain of responsibility pattern for the transport pipeline. If we're talking about all the delegating handlers, it is more like a decorator. If we're talking about only the transport delegating handlers, it isn't reall CoR either, as the one that is desired is created rather than all being created and the first one to handle it terminates the chain.

Reply:
Seems like we aren't following any one pattern, so I'll remove the reference itself.